File: infra/protocol/go/codec/formats/ros2.go
package formats

func ROS2MsgFormat(typeName string) DataFormat {
	return DataFormat{
		Kind:     DataFormatROS2Msg,
		TypeName: typeName,
	}
}

func ROS2SrvFormat(typeName string) DataFormat {
	return DataFormat{
		Kind:     DataFormatROS2Srv,
		TypeName: typeName,
	}
}

func ROS2IDLFormat(typeName string) DataFormat {
	return DataFormat{
		Kind:     DataFormatROS2IDL,
		TypeName: typeName,
	}
}

func ROS2TypeSupportFormat(typeName string) DataFormat {
	return DataFormat{
		Kind:     DataFormatROS2TypeSupport,
		TypeName: typeName,
	}
}

File: infra/protocol/python/pacific_rim_protocol/formats/ros2.py
from __future__ import annotations

from .base import DataFormat, DataFormatKind


def ros2_msg_format(type_name: str, *, package: str = "", schema_path: str = "") -> DataFormat:
  return DataFormat(
    kind=DataFormatKind.ROS2_MSG,
    type_name=type_name,
    package=package,
    schema_path=schema_path,
  )


def ros2_srv_format(type_name: str, *, package: str = "", schema_path: str = "") -> DataFormat:
  return DataFormat(
    kind=DataFormatKind.ROS2_SRV,
    type_name=type_name,
    package=package,
    schema_path=schema_path,
  )


def ros2_idl_format(type_name: str, *, package: str = "", schema_path: str = "") -> DataFormat:
  return DataFormat(
    kind=DataFormatKind.ROS2_IDL,
    type_name=type_name,
    package=package,
    schema_path=schema_path,
  )


def ros2_type_support_format(type_name: str, *, package: str = "", schema_path: str = "") -> DataFormat:
  return DataFormat(
    kind=DataFormatKind.ROS2_TYPE_SUPPORT,
    type_name=type_name,
    package=package,
    schema_path=schema_path,
  )
